subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -681,7 +684,7 @@ static bool valid_register_param(const struct bt_csip_set_member_register_param
 		return false;
 	}
 
-	if (param->rank > 0 && param->rank > param->set_size) {
+	if (param->rank > 0 && param->set_size > 0 && param->rank > param->set_size) {
 		LOG_DBG("Invalid rank: %u (shall be less than set_size: %u)", param->set_size,
```

Comment:
first arg should be param->rank and should it be 'less than' or 'less than or equal', which seems to be what the test checks for